src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8703,15 +8703,22 @@ private TypeWithAnnotations GetDeclaredParameterResult(ParameterSymbol parameter
             // after indices have been visited, and only if the receiver has not changed.
             _ = CheckPossibleNullReceiver(receiver);
 
+            VisitRvalue(node.LengthOrCountAccess);
+
             VisitRvalue(node.Argument);
-            var patternSymbol = node.PatternSymbol;
-            if (receiverType is object)
-            {
-                patternSymbol = AsMemberOfType(receiverType, patternSymbol);
-            }
+            VisitRvalue(node.IndexerAccess);
```

Comment:
>VisitRvalue [](http://example.com/codeflow?start=12&length=11)

Can ```BoundIndexOrRangePatternIndexerAccess``` be used as an Lvalue?

Reply:
Adding test for that (`ImplicitIndexerAccessAsLValue`)